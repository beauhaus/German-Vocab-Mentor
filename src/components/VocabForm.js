import React, { Component } from 'react'


class VocabForm extends Component {
    state = {
        word_de: '',
        example_de: '',
        word_en: '',
        example_en: '',
        note: '',
        isShowing: true
    }

    onWord_deChange = (e) => {
        const word_de = e.target.value;
        this.setState(() =>({word_de}))
    }
    onExample_deChange = (e) => {
        const example_de = e.target.value;
        this.setState(() =>({example_de}))
    }
    onWord_enChange = (e) => {
        const word_en = e.target.value;
        this.setState(() =>({word_en}))
    }
    onExample_enChange = (e) => {
        const example_en = e.target.value;
        this.setState(() =>({example_en}))
    }
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() =>({note}))
    }
    onIsShowingChange = (e) => {
        const isShowing = e.target.value;
        this.setState(() =>({isShowing}))
    }

    render() {
        return (
            <div>
                <form>
                    <input
                        type="text"
                        placeholder="word_de"
                        autofocus
                        value={this.state.word_de}
                    />
                    <input
                        type="text"
                        placeholder="word_en"
                        autofocus
                        value={this.state.word_en}
                    />
                    <input
                        type="text"
                        placeholder="example_de"
                        autofocus
                        value={this.state.example_de}
                    />
                    <input
                        type="text"
                        placeholder="example_en"
                        autofocus
                        value={this.state.example_en}
                    />
                    <input
                        type="text"
                        placeholder="amount"
                    />
                    <br />
                    <textarea placeholder="Add Note"></textarea>
                    <br />
                    <button type="submit">Add VocabItem</button>
                </form>
            </div>
        )
    }

}

export default VocabForm;