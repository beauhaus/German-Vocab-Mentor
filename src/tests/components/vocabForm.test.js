import React from 'react';
import { shallow } from 'enzyme';
import VocabForm from '../../components/VocabForm';
import VocabItems from '../fixtures/vocabItems';
import { wrap } from 'module';


test('renders expense form properly', () => {
    const wrapper = shallow(<VocabForm />);
    expect(wrapper).toMatchSnapshot()
})

test('should render VocabForm with VocabForm Data', () => {
    const wrapper = shallow(<VocabForm vocabItem={VocabItems[1]} />);
    expect(wrapper).toMatchSnapshot()
})

test('should render Error with vocabform for invalid submission', () => {
    const wrapper = shallow(<VocabForm />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('form').simulate('submit', {
        preventDefault: () => { }
    })
    expect(wrapper.state('error').length).toBeGreaterThan(0);
    expect(wrapper).toMatchSnapshot();
})

test('should set German word (word_de) on input change', () => {
    const value = 'NewGermanWord';
    const wrapper = shallow(<VocabForm />);
    wrapper.find('input').at(0).simulate('change', {
        target: { value }
    })
    expect(wrapper.state('word_de')).toBe(value)
})

test('should set English word (word_en) on input change', () => {
    const value = 'NewEnglishWord';
    const wrapper = shallow(<VocabForm />);
    wrapper.find('input').at(1).simulate('change', {
        target: { value }
    })
    expect(wrapper.state('word_en')).toBe(value)
})

test('should set German Sentence (example_de) on input change', () => {
    const value = 'German Sentence Example';
    const wrapper = shallow(<VocabForm />);
    wrapper.find('input').at(2).simulate('change', {
        target: { value }
    })
    expect(wrapper.state('example_de')).toBe(value)
})

test('should set English Sentence (example_en) on input change', () => {
    const value = 'English Sentence Example';
    const wrapper = shallow(<VocabForm />);
    wrapper.find('input').at(3).simulate('change', {
        target: { value }
    })
    expect(wrapper.state('example_en')).toBe(value)
})

test('should set difficulty (1-10) on input change', () => {
    const value = '7';
    const wrapper = shallow(<VocabForm />);
    wrapper.find('input').at(4).simulate('change', {
        target: { value }
    })
    expect(wrapper.state('difficulty')).toBe(value)
})


test('should set difficulty (1-10) on input change', () => {
    const value = '7';
    const wrapper = shallow(<VocabForm />);
    wrapper.find('input').at(4).simulate('change', {
        target: { value }
    })
    expect(wrapper.state('difficulty')).toBe(value)
})

test('should set isShowing on input change', () => {
    const value = false;
    const wrapper = shallow(<VocabForm />);
    wrapper.find('input').at(5).simulate('change', {
        target: { value }
    })
    expect(wrapper.state('isShowing')).toBe(value)
})

test('should show not on text area change', () => {
    const value = "New Note value";
    const wrapper = shallow(<VocabForm />);
    wrapper.find('textarea').simulate('change', {
        target: { value }
    })
    expect(wrapper.state('note')).toBe(value)
})


