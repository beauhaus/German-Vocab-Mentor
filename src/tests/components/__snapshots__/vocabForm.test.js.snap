// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders expense form properly 1`] = `
<div
  className="vocab-form"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="German Word"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Eng Word"
      type="text"
      value=""
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="German Sentence"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="English Sentence"
      type="text"
      value=""
    />
    <hr />
    <label
      htmlFor="difficulty"
    >
      difficulty:
    </label>
    <input
      max="10"
      min="1"
      name="difficulty"
      onChange={[Function]}
      placeholder="1"
      type="number"
      value={1}
    />
    <input
      checked={true}
      name="isShowing"
      onChange={[Function]}
      type="checkbox"
    />
    <textarea
      onChange={[Function]}
      placeholder="Add Note"
    />
    <button
      type="submit"
    >
      Add VocabItem
    </button>
  </form>
</div>
`;

exports[`should render Error with vocabform for invalid submission 1`] = `
<div
  className="vocab-form"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="German Word"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Eng Word"
      type="text"
      value=""
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="German Sentence"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="English Sentence"
      type="text"
      value=""
    />
    <hr />
    <label
      htmlFor="difficulty"
    >
      difficulty:
    </label>
    <input
      max="10"
      min="1"
      name="difficulty"
      onChange={[Function]}
      placeholder="1"
      type="number"
      value={1}
    />
    <input
      checked={true}
      name="isShowing"
      onChange={[Function]}
      type="checkbox"
    />
    <textarea
      onChange={[Function]}
      placeholder="Add Note"
    />
    <button
      type="submit"
    >
      Add VocabItem
    </button>
  </form>
</div>
`;

exports[`should render Error with vocabform for invalid submission 2`] = `
<div
  className="vocab-form"
>
  <p
    className="form-error"
  >
    Please enter words
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="German Word"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Eng Word"
      type="text"
      value=""
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="German Sentence"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="English Sentence"
      type="text"
      value=""
    />
    <hr />
    <label
      htmlFor="difficulty"
    >
      difficulty:
    </label>
    <input
      max="10"
      min="1"
      name="difficulty"
      onChange={[Function]}
      placeholder="1"
      type="number"
      value={1}
    />
    <input
      checked={true}
      name="isShowing"
      onChange={[Function]}
      type="checkbox"
    />
    <textarea
      onChange={[Function]}
      placeholder="Add Note"
    />
    <button
      type="submit"
    >
      Add VocabItem
    </button>
  </form>
</div>
`;

exports[`should render VocabForm with VocabForm Data 1`] = `
<div
  className="vocab-form"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="German Word"
      type="text"
      value="Ich sehne mich nach"
    />
    <input
      onChange={[Function]}
      placeholder="Eng Word"
      type="text"
      value="I long for/ I crave"
    />
    <br />
    <input
      onChange={[Function]}
      placeholder="German Sentence"
      type="text"
      value="Ich sehne mich nach Glühwein"
    />
    <input
      onChange={[Function]}
      placeholder="English Sentence"
      type="text"
      value="I'm craving Glühwein"
    />
    <hr />
    <label
      htmlFor="difficulty"
    >
      difficulty:
    </label>
    <input
      max="10"
      min="1"
      name="difficulty"
      onChange={[Function]}
      placeholder="1"
      type="number"
      value="5"
    />
    <input
      checked={true}
      name="isShowing"
      onChange={[Function]}
      type="checkbox"
    />
    <textarea
      onChange={[Function]}
      placeholder="Add Note"
    />
    <button
      type="submit"
    >
      Add VocabItem
    </button>
  </form>
</div>
`;
